iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp

```diff
@@ -307,6 +307,318 @@ class AdjustDepthwiseFilterShape : public OpRewritePattern<mhlo::ConvOp> {
   }
 };
 
+// Rewrites rank-3 mhlo.dot general in a form where lhs contraction is 2 and rhs
```

Comment:
Could you clean up this comment please. I am not sure what '2' and '1' are. Just some context would be good (I am not totally aware of all the attributes on dot general)